wom/src/main/scala/wom/types/WomIntegerLike.scala

```diff
@@ -0,0 +1,24 @@
+package wom.types
+
+
+object WomIntegerLike {
+  private def parseString(string: String): Long = {
+    if (string.contains("e") || string.contains("E")) {
+      // More flexible integer parsing that can handle the sort of scientific notation Circe likes to produce
+      // but is not as precise as `toLong` so only use if needed.
+      java.lang.Double.valueOf(string).longValue()
+    } else {
+      string.toLong
+    }
```

Comment:
Yeah the gross got grosser here. There was a loss of precision in `Double.valueOf`ing a String unnecessarily that did not escape the notice of one of our unit tests. Consequently this is now more defensive and only invokes `Double.valueOf` if it appears scientific notation might be in use.

Reply:
This is bugging me too. The way this code has "evolved" I now think it would be cleaner to do this with Circe's `fold` back in in the `CwlPreProcessor`; let me see if I can get that to work.